src/services/formValidator.js

```diff
@@ -0,0 +1,99 @@
+const INPUT_TYPES = {
+  DATE: 'date',
+};
+
+const checkInputValid = (control, controls) => {
+  const { value, validation, type } = control;
+  if (!validation.isRequired && !value) {
+    return {
+      isValid: true,
+    };
+  }
+
+  const checkValue = value.trim();
+  let isValid = true;
+  let errMessage = 'Required field';
+  if (validation.minLength) {
+    isValid = checkValue.length >= validation.minLength && isValid;
+    if (!isValid) {
+      errMessage = `Must contains at least ${validation.minLength} symbols`;
+    }
+  }
+  if (validation.lessThan) {
+    if (type === INPUT_TYPES.DATE) {
+      const { controlName } = validation.lessThan;
+      if (controlName) {
+        const anotherControlValue = controls[controlName].value;
+        if (anotherControlValue) {
+          isValid = new Date(value).getTime() <= new Date(anotherControlValue).getTime() && isValid;
+        }
+      } else {
+        isValid = new Date(value).getTime() <= validation.lessThan.value && isValid;
+      }
+    } else {
+      isValid = value <= validation.lessThan.value && isValid;
+    }
+    if (!isValid) {
+      errMessage = validation.lessThan.errMessage;
+    }
+  }
+  if (validation.moreThan) {
+    if (type === INPUT_TYPES.DATE) {
+      const { controlName } = validation.moreThan;
+      if (controlName) {
+        const anotherControlValue = controls[controlName].value;
+        if (anotherControlValue) {
+          isValid = new Date(value).getTime() >= new Date(anotherControlValue).getTime() && isValid;
+        }
+      } else {
+        isValid = new Date(value).getTime() <= validation.moreThan.value && isValid;
+      }
+    } else {
+      isValid = value >= validation.moreThan.value && isValid;
+    }
+    if (!isValid) {
+      errMessage = validation.moreThan.errMessage;
+    }
+  }
+  if (validation.email) {
+    isValid =
+      String(value)
+        .toLowerCase()
+        .match(
+          /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+        ) && isValid;
+
+    if (!isValid) {
+      errMessage = `Incorrect email`;
+    }
+  }
+
+  if (validation.phoneNumber) {
+    isValid =
+      String(value)
+        .toLowerCase()
+        .match(/^(\+375|80)(29|25|44|33)(\d{3})(\d{2})(\d{2})$/) && isValid;
```

Comment:
`/^(\+375|80)(29|25|44|33)(\d{3})(\d{2})(\d{2})$/` - regExp pattern should be saved in constant

Reply:
I saved it in constant.
```suggestion
const phoneValidationRegEx = /^(\+375|80)(29|25|44|33)(\d{3})(\d{2})(\d{2})$/;
```